src/folding/nova/circuits.rs

```diff
@@ -7,30 +12,35 @@ use ark_r1cs_std::{
     fields::fp::FpVar,
     groups::GroupOpsBounds,
     prelude::CurveVar,
+    ToConstraintFieldGadget,
 };
 use ark_relations::r1cs::{Namespace, SynthesisError};
 use core::{borrow::Borrow, marker::PhantomData};
 
 use super::CommittedInstance;
-use crate::folding::circuits::cyclefold::ECRLC;
+use crate::folding::circuits::{cyclefold::ECRLC, nonnative::NonNativeAffineVar};
 
 /// CF1 represents the ConstraintField used for the main Nova circuit which is over E1::Fr.
 pub type CF1<C> = <<C as CurveGroup>::Affine as AffineRepr>::ScalarField;
 /// CF2 represents the ConstraintField used for the CycleFold circuit which is over E2::Fr=E1::Fq.
 pub type CF2<C> = <<C as CurveGroup>::BaseField as Field>::BasePrimeField;
 
-/// CommittedInstance on E1 contains the u and x values which are folded on the main Nova
+/// CommittedInstance on E1 contains the u, x, cmE and cmW values which are folded on the main Nova
```

Comment:
If we're removing E1 from everywhere, we should do so in the comments too no?
```suggestion
/// CommittedInstance the curve's Scalar Field contains the u, x, cmE and cmW values which are folded on the main Nova
```

Reply:
Right, the comment was a bit confusing. The `E1` and `E2` refer to the main curve and the auxiliary curve of the cycle of curves used in CycleFold. I've updated this and other comments to state it more clear: https://github.com/privacy-scaling-explorations/folding-schemes/compare/57cad9fbaeda9ba1d097d385f0c151b4bb3b5e73..042fce73bbec0207a670d7ce00b1e38c9fe81d14

Reply:
Hmmmm. I thought we might be able to make this generic and curve independent. But I realized not as we need to access BaseField or ScalarField depending on the curve we're at. 

So apologies. My bad on the suggestion.
Feel free to merge!

Reply:
Nono, but you were right on that the comments were not clear. Now it is more detailed I think, thanks for the comment ^^